### PR TITLE
Add controller and route for RA location

### PR DESCRIPTION
**Pull request by arothuis**
[116602125](https://www.pivotaltracker.com/story/show/116602125)

### CODE REVIEW

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+use Surfnet\Stepup\Configuration\Value\Institution;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Query\RaLocationQuery;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Service\RaLocationService;
+use Surfnet\StepupMiddleware\ApiBundle\Response\JsonCollectionResponse;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+
+final class RaLocationController extends Controller
+{
+    public function getAction(Request $request, Institution $institution)
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php]: What about `listAction`?

File: src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml [hunk 2]
```diff
@@ -123,3 +123,11 @@ ra_candidate:
     requirements:
         identityId: '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
     condition: "request.headers.get('Accept') matches '/^application\\\\/json($|[;,])/'"
+
+ra_location:
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml]: `ra_location_listing`?

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public function listAction(Request $request, Institution $institution)
+    {
+        $this->denyAccessUnlessGranted(['ROLE_RA', 'ROLE_SS']);
+
+        $query                 = new RaLocationQuery();
+        $query->institution    = $institution;
+        $query->orderBy        = $request->get('orderBy', $query->orderBy);
+        $query->orderDirection = $request->get('orderDirection', $query->orderDirection);
+
+        $raLocations = $this->getRaLocationService()->findByInstitution($query);
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php]: The other services call this `search` (or a variant thereof).

Other than that, good work :ok_hand:, :ship:!

File: src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+final class RaLocationController extends Controller
+{
+    public function listAction(Request $request, Institution $institution)
+    {
+        $this->denyAccessUnlessGranted(['ROLE_RA', 'ROLE_SS']);
+
+        $query                 = new RaLocationQuery();
+        $query->institution    = $institution;
+        $query->orderBy        = $request->get('orderBy', $query->orderBy);
+        $query->orderDirection = $request->get('orderDirection', $query->orderDirection);
```
Comment (relaxnow) [inline src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php]: No defaults?

Comment (relaxnow) [inline src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php]: Also ```findByInstitution``` does not appear to do anything with orderBy and orderByDirection yet?

Comment (arothuis) [inline src/Surfnet/StepupMiddleware/ApiBundle/Controller/RaLocationController.php]: Good catch, not sure why I didn't implement the ordering yet. Will do.

**Comment by arothuis** [comment]
@rjkip @relaxnow

**Comment by rjkip** [comment]
Oooh, caaan dooo!!

:+1:

**Pull request merged**